openedx/edx-ora2
# Jkantor/esg test data mgmt

**TL;DR -**  Add management command for setup of tests for ORA and ESG testing 

JIRA: [AU-462](https://openedx.atlassian.net/browse/AU-462)

**What changed?**

Added new management command, `create_oa_submissions_from_file`. This management command must be run from within the LMS shell after locally installing ORA.

it takes the form `./manage.py lms create_oa_submissions_from_file <course_id> <config_file_location> [--init] [--submit] [--reset]`

course id  is the target course
config file location is where submissions-config is, relative to the edx-ora folder
--init --submit and --reset are the three modes. they're mutually exclusive and one is always required.

--init goes through the config file and creates all mentioned users. usernames that submit are created and enrolled as learners. usernames mentioned in "lock" or "gradedBy" are created and enrolled as staff. --init then continues into the logic for --submit.

--submit creates a submission for every entry in the "submissions" list for each specified ORA for the given submitter. Currently, submissions are the username of the submitter, a line break, and some sentences of lorem ipsum (due to a bug in the lorem ipsum library, they look pretty janky for now). Then, if the submission has gradeData, we create a staff assessment with the given criteria-option selections and feedback.

--reset clears all submissions. Rather than clearing all users for every ORA in the course, it looks at the config file and clears all learners for all oras they appear under.

**Example config file**

```json
[
  {
    "displayName": "My ORA",
    "submissions": [
      {
        "username": "user-1",
        "lockOwner": null,
        "gradeData":{
            "gradedBy": "edx",
            "score": {
              "pointsEarned": 5,
              "pointsPossible": 10
            },
            "overallFeedback": "Lore the ipsum",
            "criteria": [
              {
                "name": "Ideas",
                "selectedOption": "Fair",
                "feedback": "some feedback for ideas"
              },
              {
                "name": "Content",
                "selectedOption": "Poor",
                "feedback": "feedback for content"
              }
            ]
        }
      },
     {
        "username": "user-2",
        "lockOwner": "edx",
        "gradeData": {}
      }
    ]
  }
]
```

**Developer Checklist**


**Testing Instructions**

#. Create a course with one or more ORAs. 
#. Modify the example input file and save it somewhere in your edx-ora2 folder.
#. Install ORA locally by running lms and studio, then calling make install-local-ora from your ORA directory.
#. From LMS shell, run the management command with your course id, file location, and the --init flag
#. All of your users should be created and enrolled, and all specified submissions, locks, and assessments should be created.
#. From LMS shell, run the management command with your course id, file location, and the --reset flag
#. All submissions, locks, and assessments should be removed.
#. Modify values in the config file, but don't introduce any new usernames. From LMS shell, run the management command with your course id, file location, and the --submit flag
# Again, all submissions, assessment, and locks should be created.

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:


FYI: @edx/masters-devs-gta
